core/src/main/scala/org/apache/spark/launcher/LauncherBackend.scala

```diff
@@ -71,6 +110,9 @@ private[spark] abstract class LauncherBackend {
     if (connection != null && lastState != state) {
       connection.send(new SetState(state))
       lastState = state
+      if(!_isConnected && stopFlag) {
```

Comment:
This code is invoked, when childProcess is launched with `LauncherServer` going down. test case scenario 1 - `startApplication` with request to `stopIfInterrupted`